modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/StructuredContentUtil.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.delivery.internal.dto.v1_0.util;
+
+import com.liferay.dynamic.data.mapping.io.DDMFormValuesSerializer;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.storage.DDMFormValues;
+import com.liferay.dynamic.data.mapping.util.DDM;
+import com.liferay.dynamic.data.mapping.validator.DDMFormValuesValidator;
+import com.liferay.headless.delivery.dto.v1_0.StructuredContent;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.journal.util.JournalConverter;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.util.LocaleThreadLocal;
+import com.liferay.portal.kernel.util.LocaleUtil;
+
+import java.io.Serializable;
+
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ *@author Luis Miguel Barcos
+ */
+public class StructuredContentUtil {
+
+	public static String createJournalArticleContent(
+			DDMFormValuesSerializer ddmFormValuesSerializer, DDM ddm,
+			DDMFormValues ddmFormValues,
+			DDMFormValuesValidator ddmFormValuesValidator,
+			DDMStructure ddmStructure, JournalConverter journalConverter)
+		throws Exception {
+
+		DDMFormValuesUtil.validateDDMFormValues(
+			ddmFormValuesValidator, ddmFormValues);
+
+		Locale originalSiteDefaultLocale =
+			LocaleThreadLocal.getSiteDefaultLocale();
+
+		try {
+			LocaleThreadLocal.setSiteDefaultLocale(
+				LocaleUtil.fromLanguageId(ddmStructure.getDefaultLanguageId()));
+
+			ServiceContext serviceContext = new ServiceContext();
+
+			serviceContext.setAttribute(
+				"ddmFormValues",
+				DDMFormValuesUtil.toText(
+					ddmFormValuesSerializer, ddmStructure.getDDMForm(),
+					ddmFormValues.getDDMFormFieldValues()));
+
+			return journalConverter.getContent(
+				ddmStructure,
+				ddm.getFields(ddmStructure.getStructureId(), serviceContext),
+				ddmStructure.getGroupId());
+		}
+		finally {
+			LocaleThreadLocal.setSiteDefaultLocale(originalSiteDefaultLocale);
+		}
+	}
+
+	public static Map<String, Serializable> getExpandoBridgeAttributes(
```

Comment:
Too little gain for encapsulating this, right?

Reply:
Yes, I changed it